internal/config/config.go

```diff
@@ -288,6 +282,9 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
 
 	for _, p := range resources.LegacyAddressPools {
 		pool, err := addressPoolFromLegacyCR(p, communities)
+		if errors.As(err, &TransientError{}) {
```

Comment:
no need to do this. The Transient check is done in the validator.
If your concern is that the error is replaced by fmt.Errorf, use `errors.Wrapf`. See https://go.dev/blog/go1.13-errors 
"When operating on wrapped errors, however, these functions consider all the errors in a chain."

Reply:
fixed. thanks for the reference.